src/main/cpp/t04_students.cpp

```diff
@@ -29,10 +29,40 @@
 
 #include "t04_students.h"
 #include <iostream>
-
+#include <vector>
+#include <algorithm>
+#include <string>
 using namespace std;
 
+struct student {
+ string firstname, secondname;
+    int mark;
+};
+
+bool cmp ( student a, student b) {
+  return a.mark > b.mark;  
+}
+
 int t04_students() {
-    
+  int n, i, mark1, mark2, mark3;
+  string temp1, temp2;
+  int mid;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The scope of the variable 'mid' can be reduced.](https://app.codacy.com/app/thejerome/IFMO_CPP_programming_20190915/pullRequest?prid=4513205)